endpoint.go

```diff
@@ -425,7 +428,7 @@ func (ep *endpoint) sbJoin(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if sb.needDefaultGW() {
+	if !ep.joinInfo.disableGatewayService && sb.needDefaultGW() {
```

Comment:
I think it is better to have this check contained inside `needDefaultGW()`, the function name says it all ;-)